### PR TITLE
cromwomify to develop: one giant diffed commit edition

### PR DESCRIPTION
**Pull request by mcovarr**
I tried multiple times doing this as a normal rebase and failed for reasons we can tech talk when I'm in next.  I may try to split this PR into multiple commits to restore some measure of proper attribution, or maybe not if it doesn't seem worth it.

### CODE REVIEW

**Comment by kshakir** [comment]
No idea what dragons you fought with to try-and-rebase. Also not sure what the ramifications for a failed-rebase means, i.e. is any code being dropped?

Anyway this code looks pretty clean to me. 👍

**Comment by cjllanwarne** [comment]
Assuming:
- This is the just the overall diff between `develop` and the current `cromwomification` branch
- Your Cromwell branch uses the build result of this, and passes unit and centaur tests
- You haven't accidentally "diffed-out" recent changes in the WDL4S develop branch
- There isn't too much more ugly reordering still to go in wdl4s
Then:
- This is awesome and IMO should be merged ASAP to avoid further divergence
- 👍

**Comment by cjllanwarne** [comment]
Pullapprove seems to ignore bullets?

👍

**Comment by cjllanwarne** [comment]
👍

**Comment by cjllanwarne** [comment]
👍

File: wom/src/test/scala/wdl4s/wdl/values/WdlValueSpec.scala [hunk 1]
```diff
@@ -42,28 +42,6 @@ class WdlValueSpec extends FlatSpec with Matchers {
...
-      val wdlType = wdlValue.wdlType
-      val fromRawString = wdlType.fromWdlString(toRawString)
-      // Test that this is a special conversion, and is not
-      // expected to be equal after a round-trip conversion.
-      fromRawString shouldNot be(wdlValue)
-      validateFloat(fromRawString.value) should be(true)
-      fromRawString.wdlType should be(wdlType)
-    }
-  }
-
```
Comment (Horneth) [inline wom/src/test/scala/wdl4s/wdl/values/WdlValueSpec.scala]: Just curious, is there a reason for this to have disappeared ?

**Comment by Horneth** [comment]
LGTM 👍

File: wom/src/test/scala/wdl4s/wdl/values/WdlValueSpec.scala (hunk 1 above)
Comment (kshakir) [inline wom/src/test/scala/wdl4s/wdl/values/WdlValueSpec.scala]: Feel free to continue discussion here, but prior discussion: https://github.com/broadinstitute/wdl4s/pull/135#discussion_r126708085

Comment (Horneth) [inline wom/src/test/scala/wdl4s/wdl/values/WdlValueSpec.scala]: Oh ok thanks ! In that case 👍

**Comment by mcovarr** [comment]
@cjllanwarne excellent checklist.  I'll check off all of those items except perhaps the guarantee of no more ugly reordering to come because who knows.  😦

**Pull request merged**